doc/sql.txt

```diff
@@ -0,0 +1,436 @@
+
+
+    Parameters: ~
+        {params} (table)  params to be bind to {statement}, it can be a list
+                          or dict
+
+    Return: ~
+        boolean or table
+
+    Usage: ~
+        db:eval("drop table if exists todos")
+        db:eval("select * from todos where id = ?", 1)
+        db:eval("insert into t(a, b) values(:a, :b)", {a = "1", b = 2021})
+
+
+sql:isopen()                                                    *sql:isopen()*
+    Predict returning true if db connection is active.
+
+
+    Return: ~
+        boolean: true if db is opened, otherwise false.
+
+
+sql:isclose()                                                  *sql:isclose()*
+    Predict returning true if db connection is deactivated.
+
+
+    Return: ~
+        boolean: true if db is close, otherwise false.
+
+
+sql:status()                                                    *sql:status()*
+    Returns current connection status Get last error code
+
+
+
+sql:exists({name})                                              *sql:exists()*
+    Check if a table with {name} exists in sqlite db
+
+
+    Parameters: ~
+        {name} (string)  the table name.
+
+    Return: ~
+        boolean
+
+
+sql:schema({tbl}, {info})                                       *sql:schema()*
+    get sql table {name} schema, if table doesn't exist then return empty
+    table.
+
+
+    Parameters: ~
+        {tbl}  (string)   the table name
+        {info} (boolean)  whether to return table info. default false.
+
+    Return: ~
+        table: list of keys or keys and their type.
+
+
+sql:create({name}, {schema})                                    *sql:create()*
+    Create a new sqlite db table with {name} based on {schema}. if
+    {schema.ensure} then create only when it doesn't exists. similar to 'create
+    if not exists'
+
+
+    Parameters: ~
+        {name}   (string)  table name
+        {schema} (table)   the table keys/column and their types
+
+    Return: ~
+        boolean
+
+    Usage: ~
+        `db:create("todos", {id = {"int", "primary", "key"}, title = "text"})`
+
+
+sql:drop({name})                                                  *sql:drop()*
+    Create a new sqlite db table with {name} based on {schema}. if
+    {schema.ensure} then create only when it doesn't exists. similar to 'create
+    if not exists'
+
+
+    Parameters: ~
+        {name} (string)  table name
+
+    Return: ~
+        boolean
+
+    Usage: ~
+        `db:drop("todos")`
+
+
+sql:insert()                                                    *sql:insert()*
+    Insert to lua table into sqlite database table.
+
+
+    Return: ~
+        boolean|integer: true incase the table was inserted successfully, and
+        the last inserted row id.
+
+    Usage: ~
+        db:insert("todos", { title = "new todo" })
+
+
+sql:update({tbl}, {specs})                                      *sql:update()*
+    Update table row with where closure and list of values
+
+
+    Parameters: ~
```

Comment:
tbl is a ambiguous, must be changed to tbl_name instead, and same goes for other functions